src/IR/TypeInference.hs

```diff
@@ -240,4 +241,4 @@ ann2Class (Ann.TBuiltin bty) = case bty of
 ann2Class (Ann.TCon tid tys) = do
   t <- mapM anns2Class tys
   return $ Classes.TCon tid t
-ann2Class (Ann.TVar tidx   ) = return $ Classes.TVar tidx
+ann2Class (Ann.TVar _tid   ) = throwError $ Compiler.UnexpectedError $ fromString "Do not know how to translate type variable yet"
```

Comment:
We will fix this in #60 when adding support for (quantified) type variable syntax